Marlin/src/core/serial_base.h

```diff
@@ -22,12 +22,29 @@
 #pragma once
 
 #include "../inc/MarlinConfigPre.h"
-#include "macros.h"
 
 #if ENABLED(EMERGENCY_PARSER)
   #include "../feature/e_parser.h"
 #endif
 
+// Used in multiple places
+// You can build it but not manipulate it.
+// There are only few places where it's required to access the underlying member: GCodeQueue, SerialMask and MultiSerial
+struct serial_index_t {
+  // A signed index, where -1 is a special case meaning no action (neither output or input)
+  int8_t  index;
+
+  // Check if the index is within the range [a ... b]
+  constexpr inline bool within(const int8_t a, const int8_t b) const { return WITHIN(index, a, b); }
```

Comment:
/!\ Beware here: Between was exclusive, the range was [a ... b[ :  b was excluded from the valid range. Using WITHIN includes b and will break the Multiserial tests. If you want to keep within, you'll need to adjust the boundary at each call site.

Reply:
Nevermind, you did it except for...